Automattic/jetpack
# Sitemaps: Fix the Hook from init to wp_loaded for Sitemap URL handler.

Fixes #

## Proposed changes:

* This PR proposes to generate the permalinks on the `wp_loaded` hook instead of the `init` hook which would allow us to get all our plugins and theme changes loaded and the modified permalinks being generated properly.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:
Steps to reproduce:

1. Setup a sitemap with active Jetpack and Sitemaps active.
2. Upload this plugin which registers a custom post type at a priority of 11 and adds it to the News Sitemap.
Plugin gist link : https://gist.github.com/rutviksavsani/4cd931d72a981b637a2b3d745ca5c359
3. Add a new post to the Artist post-type.
4. Now on checking the URL of the post on the News post-type, You should be able to see a raw URL and not a final Permalink.
5. Now when applying this PR, we should be able to see the correct URLs as we wait for all the plugins and theme actions to get registered. ( Note: Make sure to delete old Transient for new links to be showed. )

Video for reproducing:


https://github.com/Automattic/jetpack/assets/53530700/67c20d76-d94d-449e-9f87-e559a3d56290

Screenshots:

Before:
![image](https://github.com/Automattic/jetpack/assets/53530700/97c7834e-c5c9-4c1a-9665-61f143339558)


After: 

![image](https://github.com/Automattic/jetpack/assets/53530700/e17bc01b-b928-497d-ace3-0afa986d3b2b)
